test/template.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const Code = require('@hapi/code');
+const Lab = require('@hapi/lab');
+
+const Template = require('../lib/template');
+
+
+const { describe, it } = exports.lab = Lab.script();
+const { expect } = Code;
+
+
+describe('template', () => {
+
+    it('constructor throws with null character in source', () => {
+
+        const fn = () => {
+
+            new Template('\u0000');
+        };
+
+        expect(fn).to.throw();
+    });
+
+    it('isTemplate returns false when not a template', () => {
+
+        expect(Template.isTemplate(null)).to.be.false();
+    });
+
+    it('_parse will parse {} templates', () => {
+
+        const template = new Template('foo {raw} \\{esc\\}{');
+        template._parse();
```

Comment:
I believe `_parse()` is called by the constructor and can probably be omitted here if you like.